src/common/vault.py

```diff
@@ -32,36 +32,56 @@ async def get_validators_index(self) -> int:
         return await deposit_data_registry_contract.get_validators_index()
 
     async def register_single_validator(
-        self, register_call_args: list, tx_params: TxParams, harvest_params: HarvestParams | None
+        self,
+        register_call_args: list,
+        tx_params: TxParams,
+        harvest_params: HarvestParams | None,
+        register_via_vault_v2: bool = False,
     ) -> HexBytes:
-        if await self.version() == 1:
-            return await self._register_single_validator_v1(
+        version = await self.version()
+
+        if version == 1:
+            return await self._vault_register_single(
                 register_call_args,
                 tx_params,
                 harvest_params,
             )
-        return await self._register_single_validator_v2(
+
+        if register_via_vault_v2:
+            return await self._vault_register_multiple(
+                register_call_args,
+                tx_params,
+                harvest_params,
+            )
+
+        return await self._deposit_registry_register_single(
             register_call_args,
             tx_params,
             harvest_params,
         )
 
     async def register_multiple_validators(
-        self, register_call_args: list, tx_params: TxParams, harvest_params: HarvestParams | None
+        self,
+        register_call_args: list,
+        tx_params: TxParams,
+        harvest_params: HarvestParams | None,
+        register_via_vault_v2: bool = False,
     ) -> HexBytes:
-        if await self.version() == 1:
-            return await self._register_multiple_validators_v1(
+        version = await self.version()
+
+        if version == 1 or register_via_vault_v2:
+            return await self._vault_register_multiple(
                 register_call_args,
                 tx_params,
                 harvest_params,
             )
-        return await self._register_multiple_validators_v2(
+        return await self._deposit_registry_register_multiple(
             register_call_args,
             tx_params,
             harvest_params,
         )
 
-    async def _register_single_validator_v1(
+    async def _vault_register_single(
```

Comment:
I've changed naming because for vaults v2 registration could be done via vault contract (when using validators manager signature). Using v1 in function name will be confusing because vault is v2.